### PR TITLE
HDDS-10370. Recon - Handle the pre-existing missing empty containers in clusters.

### PR DESCRIPTION
**Pull request by devmadhuu**
## What changes were proposed in this pull request?
This PR addresses the corner case when a Recon has earlier identified some MISSING (UNHEALTHY) containers, but they all were EMPTY (No keys mapped), so Recon should filter out such existing MISSING (UNHEALTHY) containers.

In a running cluster, Recon may have identified some MISSING (UNHEALTHY) containers before the [HDDS-9695](https://issues.apache.org/jira/browse/HDDS-9695) was applied. After the fix is applied, in a running cluster, existing MISSING (UNHEALTHY) containers which were actually EMPTY were still shown as MISSING, so this PR change is to filter out such existing MISSING (UNHEALTHY) containers which were actually EMPTY.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-10370

## How was this patch tested?
Existing Junit integration test `org.apache.hadoop.ozone.recon.TestReconTasks#testEmptyMissingContainerDownNode` was updated and tested.

### CODE REVIEW

**Comment by devmadhuu** [comment]
@sumitagrawl @dombizita @fapifta , pls review.

**Comment by devmadhuu** [comment]
Working on fixing TestContainerHealthTask test case failure.

**Comment by devmadhuu** [comment]
> Working on fixing TestContainerHealthTask test case failure.

This is fixed.

File: hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/persistence/ContainerHealthSchemaManager.java [hunk 1]
```diff
@@ -113,6 +118,10 @@ public Cursor<UnhealthyContainersRecord> getAllUnhealthyRecordsCursor() {
   }
 
   public void insertUnhealthyContainerRecords(List<UnhealthyContainers> recs) {
+    recs.forEach(rec -> {
```
[the remark this answers is not in the document]
Comment (devmadhuu) [reply hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/persistence/ContainerHealthSchemaManager.java]: Ok, good suggestion @sodonnel . I have handled it.

File: hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/fsck/TestContainerHealthTask.java [hunk 2]
```diff
@@ -110,38 +112,86 @@ public void testRun() throws Exception {
       when(scmClientMock.getContainerWithPipeline(c.getContainerID()))
           .thenReturn(new ContainerWithPipeline(c, null));
     }
+
+    ReplicatedReplicationConfig replicationConfig = new ReplicatedReplicationConfig() {
```
[the remark this answers is not in the document]
Comment (devmadhuu) [reply hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/fsck/TestContainerHealthTask.java]: > Do we need all this? Can we not just do:
> 
> ```
> ReplicatedReplicationConfig replicationConfig = RatisReplicationConfig.getInstance(THREE);
> ```

Yes, its not needed. I added earlier due to a error. Now simplified. Thanks.

**Review by sumitagrawl — APPROVED** [review]
LGTM

**Review by sodonnel — APPROVED** [review]
LGTM

**Pull request merged**